### PR TITLE
[Docs] Translate "Model Complexity Analysis" to Chinese.

### PR DESCRIPTION
**Pull request by VoyagerXvoyagerx**
## Modification

Translate ["Model Complexity Analysis"](https://github.com/open-mmlab/mmengine/blob/main/docs/en/advanced_tutorials/model_analysis.md) to Chinese 

Best wishes to mmengine and the developers! :blue_heart:

### CODE REVIEW

**Comment by zhouzaida** [comment]
> [CLA assistant check](https://cla-assistant.io/open-mmlab/mmengine?pullRequest=969) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/open-mmlab/mmengine?pullRequest=969) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open-mmlab/mmengine?pullRequest=969) it.

Hi, thanks for your contribution. Please sign the CLA before we merge this PR.

**Comment by VoyagerXvoyagerx** [comment]
> > [CLA assistant check](https://cla-assistant.io/open-mmlab/mmengine?pullRequest=969) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/open-mmlab/mmengine?pullRequest=969) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open-mmlab/mmengine?pullRequest=969) it.
> 
> Hi, thanks for your contribution. Please sign the CLA before we merge this PR.

done!

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 1]
```diff
@@ -1,3 +1,171 @@
...
+input_shape = (1, 10)
+model = TestNet()
+
+analysis_results = get_model_complexity_info(model, input_shape)
+
+print(analysis_results['out_table'])
+print(analysis_results['out_arch'])
+
+print("Model Flops:{}".format(analysis_results['flops_str']))
+print("Model Parameters:{}".format(analysis_results['params_str']))
```
Comment (HAOCHENYE) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:57]:
```suggestion
from torch import nn

from mmengine.analysis import get_model_complexity_info


# 以字典的形式返回分析结果，包括:
# ['flops', 'flops_str', 'activations', 'activations_str', 'params', 'params_str', 'out_table', 'out_arch']
class InnerNet(nn.Module):
    def __init__(self):
        super().__init__()
        self.fc1 = nn.Linear(10, 10)
        self.fc2 = nn.Linear(10, 10)

    def forward(self, x):
        return self.fc1(self.fc2(x))
    

class TestNet(nn.Module):
    def __init__(self):
        super().__init__()
        self.fc1 = nn.Linear(10, 10)
        self.fc2 = nn.Linear(10, 10)
        self.inner = InnerNet()

    def forward(self, x):
        return self.fc1(self.fc2(self.inner(x)))


input_shape = (1, 10)
model = TestNet()
analysis_results = get_model_complexity_info(model, input_shape)
print(analysis_results['out_table'])
print(analysis_results['out_arch'])
print("Model Flops:{}".format(analysis_results['flops_str']))
print("Model Parameters:{}".format(analysis_results['params_str']))

```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 2]
```diff
@@ -1,3 +1,171 @@
...
+)
+```
+
+## 用法示例2: 通过 mmengine 构建的模型
+
+### 代码
+
+```python
+import torch.nn.functional as F
+im
```
Comment (HAOCHENYE) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:126]: typo?

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 3]
```diff
@@ -1,3 +1,171 @@
...
+print(analysis_results['out_table'])
+print(analysis_results['out_arch'])
+
+print("Model Flops:{}".format(analysis_results['flops_str']))
+print("Model Parameters:{}".format(analysis_results['params_str']))
+```
+
+### 结果描述
+
+返回的输出是一个包含以下7个键的字典:
```
Comment (HAOCHENYE) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:62]:
```suggestion
返回的 `analysis_results` 是一个包含以下 7 个键的字典:
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 4]
```diff
@@ -1,3 +1,171 @@
...
+
+- `flops`: flop 的总数, 例如, 10000, 10000
+- `flops_str`: 格式化的字符串, 例如, 1.0G, 100M
+- `params`: 全部参数的数量, 例如, 10000, 10000
+- `params_str`: 格式化的字符串, 例如, 1.0G, 100M
+- `activations`: 激活量的总数, 例如, 10000, 10000
+- `activations_str`: 格式化的字符串, 例如, 1.0G, 100M
+- `out_table`: 以表格形式打印相关信息
```
Comment (HAOCHENYE) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:70]:
```suggestion
- `flops`: flop 的总数, 例如, 1000, 1000000
- `flops_str`: 格式化的字符串, 例如, 1.0G, 1.0M
- `params`: 全部参数的数量, 例如, 1000, 1000000
- `params_str`: 格式化的字符串, 例如, 1.0K, 1M
- `activations`: 激活量的总数, 例如, 1000, 1000000
- `activations_str`: 格式化的字符串, 例如, 1.0G, 1M
- `out_table`: 以表格形式打印相关信息
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 5]
```diff
@@ -1,3 +1,171 @@
...
+### 输出
+
+```bash
+Model Flops:4.145G
+Model Parameters:25.557M
+```
+
+## 接口
+
+我们提供了更多的选项来支持自定义输出内容：
```
Comment (HAOCHENYE) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:165]: 除了上述基本用法，`get_model_complexity_info` 还能接受以下参数，输出定制化的统计结果

**Review by HAOCHENYE — COMMENTED** [review]
Thanks for your contributions :smile: !

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 6]
```diff
@@ -1,3 +1,171 @@
 # 模型复杂度分析
 
-翻译中，请暂时阅读英文文档 [Model Complexity Analysis](https://mmengine.readthedocs.io/en/latest/advanced_tutorials/model_analysis.html)。
+我们提供了一个工具来帮助进行网络的复杂性分析。我们借鉴了 [fvcore](https://github.com/facebookresearch/fvcore) 的实现思路来构建这个工具，并计划在未来支持更多的自定义运算符。目前的工具提供了用于计算给定模型的 "parameter"、"activation" 和 "flops "的接口，并支持以网络结构或表格的形式逐层打印相关信息，同时提供 operator 层级和模块级的 flop 计数。如果您对如何准确测量一个运算符的 flop 的实现细节感兴趣，请参考 [Flop Count](https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md)。
+
+## 什么是 FLOPs
```
[the remark this answers is not in the document]
Comment (VoyagerXvoyagerx) [reply docs/zh_cn/advanced_tutorials/model_analysis.md:5]: 我觉得 FLOP 更好。我仔细研究了FLOPS的全称，缩写FLOPS中的S代表 **f**loating **p**oint **o**perations per **s**econd 中的 second，所以FLOP代表 floating point operations。我之前使用FLOPs是因为我在看到的一些论文里使用了FLOPs，这里的s应该代表operation的复数。

[the remark this answers is not in the document]
Comment (VoyagerXvoyagerx) [reply docs/zh_cn/advanced_tutorials/model_analysis.md:5]: 我现在觉得用 FLOP 更好。我参考[Wiki 中 FLOPS 的定义](floating point operations per second ) ，缩写FLOPS中的S代表 **f**loating **p**oint **o**perations per **s**econd 中的 second。我之前使用FLOPs是因为我在看到的一些论文里使用了FLOPs，这里的s应该代表operation的复数。但是根据Wiki的定义，我认为用FLOP表示**f**loating **p**oint **o**perations 是可以的。

File: docs/zh_cn/advanced_tutorials/model_analysis.md (hunk 2 above)
Comment (VoyagerXvoyagerx) [reply docs/zh_cn/advanced_tutorials/model_analysis.md:128]: Sorry, I'll fix it soom.

File: docs/zh_cn/advanced_tutorials/model_analysis.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (VoyagerXvoyagerx) [reply docs/zh_cn/advanced_tutorials/model_analysis.md:5]: 1. 避免使用FLOPs，以免与FLOPS混淆  2. 考虑到代码里使用了小写的 flop 或 flops，文档里统一使用小写的 flop 或 flops 表示浮点运算量。

**Review by HAOCHENYE — APPROVED** [review]

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 7]
```diff
@@ -1,3 +1,173 @@
 # 模型复杂度分析
 
-翻译中，请暂时阅读英文文档 [Model Complexity Analysis](https://mmengine.readthedocs.io/en/latest/advanced_tutorials/model_analysis.html)。
+我们提供了一个工具来帮助进行网络的复杂性分析。我们借鉴了 [fvcore](https://github.com/facebookresearch/fvcore) 的实现思路来构建这个工具，并计划在未来支持更多的自定义运算符。目前的工具提供了用于计算给定模型的参数量（parameter）、激活量（activation） 和浮点运算量）的接口，并支持以网络结构或表格的形式逐层打印相关信息，同时提供算子（operator）级和模块级（Module）的浮点运算量统计。如果您对如何准确测量一个运算符的 flop 的实现细节感兴趣，请参考 [Flop Count](https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md)。
```
Comment (zhouzaida) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:3]:
```suggestion
我们提供了一个工具来帮助分析网络的复杂性。我们借鉴了 [fvcore](https://github.com/facebookresearch/fvcore) 的实现思路来构建这个工具，并计划在未来支持更多的自定义算子。目前的工具提供了用于计算给定模型的参数量（parameter）、激活量（activation） 和浮点运算量的接口，并支持以网络结构或表格的形式逐层打印相关信息，同时提供了算子（operator）级别和模块级别（Module）的统计。如果您对如何统计运算符 flop 的实现细节感兴趣，请参考 [Flop Count](https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md)。
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 8]
```diff
@@ -1,3 +1,173 @@
...
+
+激活量（activation）用于衡量某一层产生的特征数量。
+
+例如，给定输入尺寸 `inputs = torch.randn((1, 3, 10, 10))`，和一个输入通道为 3、输出通道为 10 的线性层 `conv = nn.Conv2d(in_channels=3, out_channels=10, kernel_size=1)`。
+
+将 `inputs` 输入到 `conv`，得到的输出特征图的尺寸是 `(1, 10, 10, 10)`。则 `output` 对这个 `conv` 层的激活量就是 `1000=10*10*10`。
+
+让我们从下面的示例开始上手。
+
+## 用法示例1: 从原始的 `nn.Module` 构建的模型
```
Comment (zhouzaida) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:19]:
```suggestion
## 用法示例1: 基于 `nn.Module` 构建的模型
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 9]
```diff
@@ -1,3 +1,173 @@
...
+    )
+    (fc2): Linear(
+      in_features=10, out_features=10, bias=True
+      #params: 0.11K, #flops: 100, #acts: 10
+    )
+  )
+)
+```
+
+## 用法示例2: 通过 mmengine 构建的模型
```
Comment (zhouzaida) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:122]:
```suggestion
## 用法示例2: 基于 BaseModel（来自 MMEngine）构建的模型
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 10]
```diff
@@ -1,3 +1,173 @@
...
+
+## 什么是浮点运算量（flop）
+
+浮点运算量（flop）在复杂性分析中不是一个定义非常明确的指标。参考 [detectron2](https://detectron2.readthedocs.io/en/latest/modules/fvcore.html#fvcore.nn.FlopCountAnalysis) 的描述，将一组乘加运算定义为 1 个 flop。
+
+## 什么是激活量（activation）
+
+激活量（activation）用于衡量某一层产生的特征数量。
+
+例如，给定输入尺寸 `inputs = torch.randn((1, 3, 10, 10))`，和一个输入通道为 3、输出通道为 10 的线性层 `conv = nn.Conv2d(in_channels=3, out_channels=10, kernel_size=1)`。
```
Comment (zhouzaida) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:13]:
```suggestion
例如，给定输入尺寸 `inputs = torch.randn((1, 3, 10, 10))`，和一个输入通道为 3、输出通道为 10 的卷积层 `conv = nn.Conv2d(in_channels=3, out_channels=10, kernel_size=1)`。
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 11]
```diff
@@ -1,3 +1,173 @@
...
+
+将 `inputs` 输入到 `conv`，得到的输出特征图的尺寸是 `(1, 10, 10, 10)`。则 `output` 对这个 `conv` 层的激活量就是 `1000=10*10*10`。
```
Comment (zhouzaida) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:15]:
```suggestion
将 `inputs` 输入到 `conv`，输出的特征图尺寸是 `(1, 10, 10, 10)`，则 `output` 对这个 `conv` 层的激活量就是 `1000=10*10*10`。
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 12]
```diff
@@ -1,3 +1,173 @@
...
+print(analysis_results['out_table'])
+print(analysis_results['out_arch'])
+print("Model Flops:{}".format(analysis_results['flops_str']))
+print("Model Parameters:{}".format(analysis_results['params_str']))
+
+```
+
+### 结果描述
+
+返回的 `analysis_results` 是一个包含以下 7 个键的字典:
```
Comment (zhouzaida) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:64]:
```suggestion
返回的 `analysis_results` 是一个包含以下 7 个信息的字典:
```

File: docs/en/advanced_tutorials/model_analysis.md [hunk 13]
```diff
@@ -1,29 +1,36 @@
...
+  Activation is used to measure the feature quantity produced from one layer.
 
-Let's start with the following examples.
+- Parameters
 
-## Usage Example 1: Model built with native nn.Module
+  The parameter count of a model.
 
-### Code
+For example, given an input size of `inputs = torch.randn((1, 3, 10, 10))` and a convolutional layer `conv = nn.Conv2d(in_channels=3, out_channels=10, kernel_size=3)`, if the output feature map size is `(1, 10, 8, 8)`, then its FLOPs are `17280 = 10*8*8*3*3*3` (where 10*8*8 represents the output feature map size, and 3*3*3 represents the computation for each output), activations are `640 = 10*8*8`, and the parameter count is `280 = 3*10*3*3 + 10` (where 3*10*3\*3 represents the size of weights, and 10 represents the size of bias).
```
Comment (HAOCHENYE) [inline docs/en/advanced_tutorials/model_analysis.md:21]:
```suggestion
For example, given an input size of `inputs = torch.randn((1, 3, 10, 10))` and a convolutional layer `conv = nn.Conv2d(in_channels=3, out_channels=10, kernel_size=3)`, if the output feature map size is `(1, 10, 8, 8)`, then its FLOPs are `17280 = 10*8*8*3*3*3` (where 10*8*8 represents the output feature map size, and 3*3*3 represents the computation for each output), activations are `640 = 10*8*8`, and the parameter count is `280 = 3*10*3*3 + 10` (where 3*10*3*3 represents the size of weights, and 10 represents the size of bias).
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 14]
```diff
@@ -1,3 +1,184 @@
 # 模型复杂度分析
 
-翻译中，请暂时阅读英文文档 [Model Complexity Analysis](https://mmengine.readthedocs.io/en/latest/advanced_tutorials/model_analysis.html)。
+我们提供了一个工具来帮助分析网络的复杂性。我们借鉴了 [fvcore](https://github.com/facebookresearch/fvcore) 的实现思路来构建这个工具，并计划在未来支持更多的自定义算子。目前的工具提供了用于计算给定模型的浮点运算量（flop）、激活量（activation）和参数量（parameter）的接口，并支持以网络结构或表格的形式逐层打印相关信息，同时提供了算子级别（operator）和模块级别（Module）的统计。如果您对统计运算符 flop 的实现细节感兴趣，请参考 [Flop Count](https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md)。
```
Comment (HAOCHENYE) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:3]:
```suggestion
我们提供了一个工具来帮助分析网络的复杂性。我们借鉴了 [fvcore](https://github.com/facebookresearch/fvcore) 的实现思路来构建这个工具，并计划在未来支持更多的自定义算子。目前的工具提供了用于计算给定模型的浮点运算量（FLOPs）、激活量（activation）和参数量（parameter）的接口，并支持以网络结构或表格的形式逐层打印相关信息，同时提供了算子级别（operator）和模块级别（Module）的统计。如果您对统计浮点运算量的实现细节感兴趣，请参考 [Flop Count](https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md)。
```

File: docs/zh_cn/advanced_tutorials/model_analysis.md [hunk 15]
```diff
@@ -1,3 +1,184 @@
...
+
+## 定义
+
+模型复杂度有 3 个指标，分别是浮点运算量（flop）、激活量（activation）以及参数量（parameter），它们的定义如下：
```
Comment (HAOCHENYE) [inline docs/zh_cn/advanced_tutorials/model_analysis.md:7]:
```suggestion
模型复杂度有 3 个指标，分别是浮点运算量（FLOPs）、激活量（activation）以及参数量（parameter），它们的定义如下：
```

**Pull request merged**